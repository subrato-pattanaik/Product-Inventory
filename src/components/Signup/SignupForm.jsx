import React from "react";
import { withRouter, useHistory } from "react-router-dom";
import { Col, Form, Button } from "react-bootstrap";
import { useForm } from "react-hook-form";
import * as yup from "yup";
import { yupResolver } from "@hookform/resolvers/yup";
import axios from "axios";

function SignupForm() {
  const schema = yup.object().shape({
    email: yup.string().required("*Email is mandatory"),
    password: yup.string().required("*Password is mandatory"),
    firstname: yup
      .string()
      .required("*First Name is mandatory")
      .matches(/^[A-Za-z]*$/, "Enter charater only")
      .max(20, "First name exceeds its length to 20"),
    lastname: yup
      .string()
      .required("*last Name is mandatory")
      .matches(/^[A-Za-z]*$/, "Enter charater only")
      .max(20, "First name exceeds its length to 20"),
    mobileno: yup
      .string()
      .required("*Mobile No is mandatory")
      .matches(/^[0-9]*$/, "Enter Number only")
      .max(10, "Must contain 10 number")
      .min(10, "Must contain 10 number"),
  });

  const { register, handleSubmit, errors } = useForm({
    resolver: yupResolver(schema),
  });

  const onSubmit = (formData) => {
    console.log(formData);

    axios.post("http://localhost:4000/users", formData);

    navigation.push({
      pathname: "/Login",
      state: { pass: "pass" },
    });
  };

  let navigation = useHistory();
  return (
    <Form autoComplete="off" onSubmit={handleSubmit(onSubmit)}>
      <Form.Row>
        <Form.Group as={Col} className="mr-2">
          <Form.Label>
            Email<span className="text-danger">*</span>
          </Form.Label>
          <Form.Control
            className="rounded-pill"
            type="email"
            placeholder="Enter email"
            name="email"
            ref={register}
          />
          {errors.email && (
            <div className="small lead text-danger mb-2">
              {errors.email.message}
            </div>
          )}
        </Form.Group>

        <Form.Group as={Col}>
          <Form.Label>
            Password<span className="text-danger">*</span>
          </Form.Label>
          <Form.Control
            className="rounded-pill"
            type="password"
            placeholder="Password"
            name="password"
            ref={register}
          />
          {errors.password && (
            <div className="small lead text-danger  mb-2">
              {errors.password.message}
            </div>
          )}
        </Form.Group>
      </Form.Row>
      <Form.Row>
        <Form.Group as={Col} className="mr-2">
          <Form.Label>
            First Name<span className="text-danger">*</span>
          </Form.Label>
          <Form.Control
            className="rounded-pill"
            placeholder="Enter First Name"
            name="firstname"
            ref={register}
          />
          {errors.firstname && (
            <div className="small lead text-danger  mb-2">
              {errors.firstname.message}
            </div>
          )}
        </Form.Group>

        <Form.Group as={Col}>
          <Form.Label>
            Last Name<span className="text-danger">*</span>
          </Form.Label>
          <Form.Control
            className="rounded-pill"
            placeholder="Enter Last Name"
            name="lastname"
            ref={register}
          />
          {errors.lastname && (
            <div className="small lead text-danger  mb-2">
              {errors.lastname.message}
            </div>
          )}
        </Form.Group>
      </Form.Row>
      <Form.Row className="mb-4">
        <Form.Group as={Col} className="mr-2">
          <Form.Label>Location</Form.Label>
          <Form.Control
            className="rounded-pill"
            placeholder="Enter Location"
            name="location"
            ref={register}
          />
        </Form.Group>

        <Form.Group as={Col}>
          <Form.Label>
            Mobile No<span className="text-danger">*</span>
          </Form.Label>
          <Form.Control
            className="rounded-pill"
            placeholder="Mobile No."
            name="mobileno"
            ref={register}
          />
          {errors.mobileno && (
            <div className="small lead text-danger  mb-2">
              {errors.mobileno.message}
            </div>
          )}
        </Form.Group>
      </Form.Row>

      <Button
        variant="warning"
        type="submit"
        className="d-block text-white w-100 rounded-pill "
        //onClick={() => navigation.push("/Login")}
      >
        Submit
      </Button>
    </Form>
  );
}

export default withRouter(SignupForm);
