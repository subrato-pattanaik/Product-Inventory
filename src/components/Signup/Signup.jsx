import React from "react";
import { withRouter } from "react-router-dom";
import { Container, Row, Col } from "react-bootstrap";
import login from "../../images/image2.jpg";
import SignupForm from "./SignupForm";

function Signup() {
  const image = {
    backgroundImage: `linear-gradient( rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3) ),url(${login})`,
    backgroundSize: "cover",
    backgroundPosition: "center center",
    minHeight: "89vh",
  };
  return (
    <Container fluid={true} className="p-0">
      <Row noGutters>
        <Col lg={6} md={6} xs={12}>
          <Container
            className="d-flex bg-light align-items-center"
            style={{ minHeight: "89vh" }}
          >
            <Row className="mx-auto p-5">
              <Col md={12} xs={12}>
                <h3 className="display-4 text-warning text-center font-weight-bold mt-n5 mb-2">
                  Sign Up
                </h3>

                <SignupForm />
              </Col>
            </Row>
          </Container>
        </Col>
        <Col lg={6} md={6} xs={12} style={image}></Col>
      </Row>
    </Container>
  );
}

export default withRouter(Signup);
