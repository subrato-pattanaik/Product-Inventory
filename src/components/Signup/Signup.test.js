import React from "react";
import { shallow, mount } from "enzyme";
import "mutationobserver-shim";
import Signup from "./Signup";
import { BrowserRouter } from "react-router-dom";
import { Form } from "react-bootstrap";

describe("Sign up Page Component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Signup.WrappedComponent />);
  });

  it("should test snapshot of Header page", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should test Form component using Full dom rendering", () => {
    const component = mount(
      <BrowserRouter>
        <Signup.WrappedComponent />
      </BrowserRouter>
    );
    expect(component.find("Form").exists()).toEqual(true);
  });

  it("should test Signup title using Full dom rendering", () => {
    const component = mount(
      <BrowserRouter>
        <Signup.WrappedComponent />
      </BrowserRouter>
    );
    expect(component.find("h3").text()).toEqual("Sign Up");
  });

  it("should contain email field there", () => {
    const component = mount(
      <BrowserRouter>
        <Signup.WrappedComponent />
      </BrowserRouter>
    );
    expect(
      component.containsMatchingElement(
        <Form.Label>
          Email<span className="text-danger">*</span>
        </Form.Label>
      )
    ).toEqual(true);
  });
});
