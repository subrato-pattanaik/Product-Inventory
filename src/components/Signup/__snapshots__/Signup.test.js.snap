// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up Page Component should test snapshot of Header page 1`] = `
<Container
  className="p-0"
  fluid={true}
>
  <Row
    noGutters={true}
  >
    <Col
      lg={6}
      md={6}
      xs={12}
    >
      <Container
        className="d-flex bg-light align-items-center"
        fluid={false}
        style={
          Object {
            "minHeight": "89vh",
          }
        }
      >
        <Row
          className="mx-auto p-5"
          noGutters={false}
        >
          <Col
            md={12}
            xs={12}
          >
            <h3
              className="display-4 text-warning text-center font-weight-bold mt-n5 mb-2"
            >
              Sign Up
            </h3>
            <withRouter(SignupForm) />
          </Col>
        </Row>
      </Container>
    </Col>
    <Col
      lg={6}
      md={6}
      style={
        Object {
          "backgroundImage": "linear-gradient( rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3) ),url(image2.jpg)",
          "backgroundPosition": "center center",
          "backgroundSize": "cover",
          "minHeight": "89vh",
        }
      }
      xs={12}
    />
  </Row>
</Container>
`;
